codecov_auth/authentication/repo_auth.py

```diff
@@ -318,3 +318,33 @@ def authenticate(self, request):
             )
         else:
             raise exceptions.AuthenticationFailed(self.auth_failed_message)
+
+
+class BundleAnalysisTokenlessAuthentication(TokenlessAuthentication):
+    def _get_info_from_request_path(
+        self, request: HttpRequest
+    ) -> tuple[Repository, str | None]:
+        try:
+            body = json.loads(str(request.body, "utf8"))
+
+            # Validate provider
+            service_enum = Service(body.get("git_service"))
+
+            # Validate that next group exists and decode slug
+            repo = get_repository_from_string(service_enum, body.get("slug"))
+            if repo is None:
+                # Purposefully using the generic message so that we don't tell that
+                # we don't have a certain repo
+                raise exceptions.AuthenticationFailed(self.auth_failed_message)
+
+            return repo, body.get("commit")
+        except json.JSONDecodeError:
+            # Validate request body format
+            raise exceptions.AuthenticationFailed(self.auth_failed_message)
+        except ValueError:
+            # Validate provider
+            raise exceptions.AuthenticationFailed(self.auth_failed_message)
+
+    def get_branch(self, request, repoid=None, commitid=None):
+        body = json.loads(str(request.body, "utf8"))
+        return body.get("branch")
```

Comment:
what should you do if the commit already exists and it has a different branch name from what was passed in the request?

the logic for coverage is:
- for `create-commit`, decide if we're allowed to create the commit based on the branch name in the request
- for `create-report` and `do-upload`, decide if we're allowed to perform the action by looking up the commit and checking the branch name it was created with

that way someone can't upload to a commit on `main` and just lie in the request and say the branch name is actually something else to bypass the check.

Reply:
Added check to make sure requested commit's branch matches what was stored in our DB